Zend/zend_execute.c

```diff
@@ -874,8 +874,13 @@ ZEND_COLD zend_never_inline void zend_magic_get_property_type_inconsistency_erro
 
 ZEND_COLD void zend_match_unhandled_error(const zval *value)
 {
+	zend_long max_len = EG(exception_string_param_max_len);
 	smart_str msg = {0};
-	if (smart_str_append_zval(&msg, value, EG(exception_string_param_max_len)) != SUCCESS) {
+	if (
+		EG(exception_ignore_args)
+		|| (Z_TYPE_P(value) == IS_STRING && max_len == 0)
+		|| smart_str_append_zval(&msg, value, max_len) != SUCCESS
+	) {
```

Comment:
I've no idea if ``type string`` is better than ``...``.
I'm aldo surprised that ``EG(exception_ignore_args)`` was introduced and wasn't handled.
@iluuu1994 @nielsdos @arnaud-lb please make a decision about this

Reply:
I think it's slightly better as it gives a little bit more information in case of mixed type matches, but I have no strong feelings to this. I don't see a technical problem.